File: backend/models/Flight.js
const mongoose = require("mongoose");

const flightSchema = new mongoose.Schema(
  {
    arrivalPlanned: { type: Date },
    arrivalExact: { type: Date },
    departurePlanned: { type: Date },
    departureExact: { type: Date },
    changeTime: { type: Number },
    status: { type: String },
    destination: { type: String },
    airport: { type: mongoose.Schema.Types.ObjectId, ref: "airport" },
    gate: { type: mongoose.Schema.Types.ObjectId, ref: "gate" },
    airline: { type: String },
    flightNumber: { type: String },
  },
  { strict: "throw" }
);

const Flight = mongoose.model("flight", flightSchema);

module.exports = Flight;

File: src/main/kotlin/model/Flight.kt
package model

import com.google.gson.annotations.SerializedName
import java.util.Date

data class Flight(
    @SerializedName("_id") val id: String? = null,
    var arrivalPlanned: Date? = null,
    var arrivalExact: Date? = null,
    val departurePlanned: Date? = null,
    val departureExact: Date? = null,
    val changeTime: Int,
    val status: String,
    val destination: String,
    val airport: String,
    var gate: String,
    val airline: String,
    val flightNumber : String,
    val destinationLat: Double? = 0.0,
    val destinationLng: Double? = 0.0,
){
    override fun toString(): String {
        return "Flight(id='$id',\n" +
                "arrivalPlanned=$arrivalPlanned,\n" +
                "arrivalExact=$arrivalExact,\n" +
                "departurePlanned=$departurePlanned,\n" +
                "departureExact=$departureExact,\n" +
                "departureChangeTime=$changeTime,\n" +
                "status='$status',\n" +
                "destination='$destination',\n" +
                "airport='$airport',\n" +
                "gate='$gate',\n" +
                "airline='$airline',\n" +
                "flightNumber='$flightNumber'\n" +
                 "destinationLat='$destinationLat',\n" +
                "destinationLng='$destinationLng')"
    }
}